### PR TITLE
Add `GlobalSortAggr` to `nn.aggr` package.

### PR DESCRIPTION
**Pull request by Padarn**
Add `GlobalSortAggr` to the new `nn.aggr` packages replace existing `global_sort_pool` from `nn.glob.sort`.

Addresses item in https://github.com/pyg-team/pytorch_geometric/issues/4712

WIP items to address:
- [x] input `dim_size` is not checked or used
- [x] input `dim` is not yet respected # skipping as to_dense_batch does not support this yet
- [x] depreciate `nn.glob` version (replace it in existing implementations?)

### CODE REVIEW

File: torch_geometric/nn/aggr/__init__.py [hunk 1]
```diff
@@ -14,20 +14,11 @@
...
-    'StdAggregation',
-    'SoftmaxAggregation',
-    'PowerMeanAggregation',
-    'LSTMAggregation',
-    'Set2Set',
-    'DegreeScalerAggregation',
+    'Aggregation', 'MultiAggregation', 'MeanAggregation', 'SumAggregation',
+    'MaxAggregation', 'MinAggregation', 'MulAggregation', 'VarAggregation',
+    'StdAggregation', 'SoftmaxAggregation', 'PowerMeanAggregation',
+    'LSTMAggregation', 'Set2Set', 'DegreeScalerAggregation', 'GlobalSortAggr'
```
Comment (rusty1s) [inline torch_geometric/nn/aggr/__init__.py:23]:
```suggestion
    'LSTMAggregation', 'Set2Set', 'DegreeScalerAggregation', 'GlobalSortAggr',
```

**Review by rusty1s — COMMENTED** [review]
Thank you! Looks amazing. If possible, let's deprecate the corresponding function in `torch_geometric.nn.glob` in this PR as well.

**Comment by Padarn** [comment]
Makes sense. Will add it to the list.

**Comment by Padarn** [comment]
Not sure why I'm getting failures in the `glob` tests - do we want to keep them?

File: torch_geometric/nn/glob/__init__.py [hunk 2]
```diff
@@ -18,8 +16,14 @@
...
 
 Set2Set = deprecated(
     details="use 'nn.aggr.Set2Set' instead",
     func_name='nn.glob.Set2Set',
 )(Set2Set)
+
+global_sort_pool = deprecated(
+    details="use 'nn.aggr.GlobalSortAggr' instead",
+    func_name='nn.glob.global_sort_pool',
+)(sort.global_sort_pool)
```
Comment (Padarn) [inline torch_geometric/nn/glob/__init__.py:29]: cannot use the new class here as its not a replacement (new class doesn't take `k` in the function call)

Comment (Padarn) [reply torch_geometric/nn/glob/__init__.py:29]: we could make it a kwarg if we want it to work exactly the same way

File: torch_geometric/nn/aggr/sort.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+from typing import Optional
+
+import torch
+from torch import Tensor
+
+from torch_geometric.nn.aggr import Aggregation
+
+
+class GlobalSortAggr(Aggregation):
+    r"""The global pooling operator from the `"An End-to-End Deep Learning
```
Comment (rusty1s) [inline torch_geometric/nn/aggr/sort.py:10]:
```suggestion
    r"""The sort pooling operator from the `"An End-to-End Deep Learning
```

File: torch_geometric/nn/aggr/sort.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+from typing import Optional
+
+import torch
+from torch import Tensor
+
+from torch_geometric.nn.aggr import Aggregation
+
+
+class GlobalSortAggr(Aggregation):
```
Comment (rusty1s) [inline torch_geometric/nn/aggr/sort.py:9]:
```suggestion
class SortAggr(Aggregation):
```
Since `Aggregation` can be used for both message passing aggregation and global pooling, let's not use `Global` inside the name. WDYT?

File: torch_geometric/nn/glob/__init__.py (hunk 2 above)
Comment (rusty1s) [reply torch_geometric/nn/glob/__init__.py:29]: I think you can just define it here:
```python
@deprecated(...)
def global_sort_pool(...):
    module = GlobalSortAggr(...)
    return module(...)
```

**Review by rusty1s — COMMENTED** [review]
I think the issue with failing tests is that there exists multiple `test_sort.py` files (which is not allowed by `pytest`). Try removing `glob/aggr/test_sort.py`.

File: torch_geometric/nn/aggr/sort.py (hunk 4 above)
Comment (Padarn) [reply torch_geometric/nn/aggr/sort.py:9]: Good point

File: torch_geometric/nn/glob/__init__.py (hunk 2 above)
Comment (Padarn) [reply torch_geometric/nn/glob/__init__.py:29]: yeah okay this is a better solution, thanks!

**Comment by Padarn** [comment]
Looks okay now. May as well remove the old version at this point. WDYT?

**Comment by Padarn** [comment]
All done :-)

**Review by lightaime — APPROVED** [review]
Thanks, @Padarn. LGTM!

**Comment by lightaime** [comment]
nit-pick: I think it may be better to rename `SortAggr` as `SortAggregation` to be consistent with other `Aggregation` classes. WDYT?

**Comment by Padarn** [comment]
I agree @lightaime! So: https://github.com/pyg-team/pytorch_geometric/pull/5085
